engine/runtime/src/main/java/org/enso/interpreter/runtime/PatchedModuleValues.java

```diff
@@ -60,18 +55,16 @@ synchronized void dispose() {
    * @param delta positive or negative change at the offset location
    */
   private synchronized void performUpdates(
-    Instrumenter instr, Map<Node, Object> collect, int offset, int delta
+    Instrumenter instr, Map<Node, Runnable> collect, int offset, int delta
   ) {
-    if (binding == null) {
-      SourceFilter sourceAssociatedWithMyModule = SourceFilter.newBuilder().
-              languageIs(LanguageInfo.ID).
-              sourceIs(module::isModuleSource).
-              build();
-      SourceSectionFilter filter = SourceSectionFilter.newBuilder().sourceFilter(sourceAssociatedWithMyModule).tagIs(Patchable.Tag.class).build();
-      this.binding = instr.attachExecutionEventListener(filter, this);
+    for (var entry : collect.entrySet()) {
+      // do the patching
+      entry.getValue().run();
```

Comment:
I guess this should be fine because we ensure that program executions are synchronous.